Makefile

```diff
@@ -47,7 +49,7 @@ SPACE:=$(subst ,, )
 SHELL:=env PATH=$(subst $(SPACE),\$(SPACE),$(PATH)) $(SHELL)
 # ==============================================================================
 # flags for reproducible go builds
-BUILD_FLAGS?=-trimpath -ldflags="-buildid="
+BUILD_FLAGS?=-trimpath -ldflags="-buildid= -X=sigs.k8s.io/kubetest2/pkg/app.GitTag=$(COMMIT)"
```

Comment:
unfortunately, handling the deployers won't be so straightforward :) 
this will just always get the kubetest2 shim version. but we can have deployers built at different commits.